server/aesdsocket.c

```diff
@@ -0,0 +1,363 @@
+	    	
+	struct addrinfo *serverInfo = NULL;
+	    
+	openlog(NULL, 0, LOG_USER);
+	
+	//check if arguments are valid with -d
+	if ((argc == 2) && (strcmp(argv[1], "-d") == 0))
+	{
+		daemonize = 1;
+		syslog(LOG_INFO, "Starting as DAEMON Process");
+	}
+	
+	//register signal handlers
+	reg_signal_handlers();
+	socketfd = socket(AF_INET, SOCK_STREAM, 0);
+	if (socketfd == -1) 
+	{
+		syslog(LOG_ERR, "ERROR: Unable to create socket");
+		cleanup(FAILURE);
+	} 
+	syslog(LOG_INFO, "socket() : Socket Created with sockfd: %d",socketfd);   	
+	    	
+	if (getaddrinfo(NULL, PORT, &hints, &serverInfo) != 0)
+	{
+		syslog(LOG_ERR, "ERROR: Unable to get socket address with getaddrinfo");
+		if (serverInfo != NULL)
+		{
+		    freeaddrinfo(serverInfo);
+		}
+		cleanup(FAILURE);
+	}
+	syslog(LOG_INFO, "getaddrinfo() : result server socket address");  	
+	
+	//Sets socket option to reuse socket address
+	int reuse = 1;
+	if (setsockopt(socketfd, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &reuse, sizeof(int)) == -1) 
+	{
+	      	syslog(LOG_ERR, "ERROR: Unable to set reuse option with setsockopt");
+	      	if (serverInfo != NULL)
+	      	{
+	      		freeaddrinfo(serverInfo);
+	      	}
+		cleanup(FAILURE);
+	}
+	syslog(LOG_INFO, "setsockopt() : Port reuse option");
+
+	//binds the socket address to file descriptor
+	if (bind(socketfd, serverInfo->ai_addr, serverInfo->ai_addrlen) != 0)
+	{
+	     	syslog(LOG_PERROR, "ERROR: Unable to bind socket and port");
+	      	if (serverInfo != NULL)
+	      	{
+	      		freeaddrinfo(serverInfo);
+	      	}
+	      	cleanup(FAILURE);
+	}
+	syslog(LOG_INFO, "bind() : Bind socket and port");
+	
+    	if (serverInfo != NULL)
+    	{
+       	freeaddrinfo(serverInfo);
+    	}
+	
+	//Start the daemon process if -d is specified
+	if (daemonize) 
+	{
+		syslog(LOG_INFO, "Running as DAEMON Process");
+       	daemonization_func();
+    	}
+	
+	if (listen(socketfd, 10) == -1) 
+	{
+       	syslog(LOG_ERR, "ERROR: unable to listen");
+       	cleanup(FAILURE);
+    	}
+	syslog(LOG_INFO, "listen() : listening for client");
+
+    	
+    	//Receives the data until a packet is completely received, write data into aesddata file location
+	while(!exit_condition_flag)
+	{
+       	char ip_addr[INET_ADDRSTRLEN];
+       	Clientfd = socket_connect(socketfd, ip_addr);
+       	if (Clientfd == -1) 
+       	{
+       		continue;
+       	}
+       	
+       	filefd = open(aesddata_file, O_CREAT | O_RDWR | O_APPEND, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+    		if (filefd == -1)
+   		{
+       		syslog(LOG_ERR, "ERROR: Unable to open file");
+       		cleanup(FAILURE);
+   		}
+   	
+       	do
+       	{
+       		memset(buf, 0, BUFF_MAX);
+       		recv_bytes = recv(Clientfd, buf, BUFF_MAX, 0);
+       		if (recv_bytes == -1)
+       		{
+           			syslog(LOG_ERR, "ERROR: unable to recieve data from client");
+      				cleanup(FAILURE);
+       		}
+
+       		written_bytes = write(filefd, buf, recv_bytes);
```

Comment:
@charan1811 with this design you will write to the file before the packet completes, which may cause problems with future assignments.